pkg/apis/machine/validation/machinedeployment_test.go

```diff
@@ -0,0 +1,183 @@
+			}))))
+		})
+
+		Context("Validate UpdateStrategy of machine deployment", func() {
+			It("should return error if strategy type is not valid", func() {
+				machineDeployment.Spec.Strategy.Type = "invalid"
+
+				Expect(ValidateMachineDeployment(machineDeployment)).To(ContainElement(PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":   Equal(field.ErrorTypeInvalid),
+					"Field":  Equal("spec.strategy.type"),
+					"Detail": Equal("strategy type must be one of [InPlaceUpdate Recreate RollingUpdate]"),
+				}))))
+			})
+
+			Context("Validate RollingUpdate strategy", func() {
+				It("should not return error if RollingUpdate strategy is valid", func() {
+					Expect(ValidateMachineDeployment(machineDeployment)).To(BeEmpty())
+				})
+
+				It("should return error if MaxUnavailable is not valid", func() {
+					machineDeployment.Spec.Strategy.RollingUpdate.UpdateConfiguration.MaxUnavailable = nil
+
+					Expect(ValidateMachineDeployment(machineDeployment)).To(ContainElement(PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":   Equal(field.ErrorTypeRequired),
+						"Field":  Equal("spec.strategy.rollingUpdate.maxUnavailable"),
+						"Detail": Equal("unable to convert maxUnavailable to int32"),
+					}))))
+				})
+
+				It("should return error if MaxSurge is not valid", func() {
+					machineDeployment.Spec.Strategy.RollingUpdate.UpdateConfiguration.MaxSurge = nil
+
+					Expect(ValidateMachineDeployment(machineDeployment)).To(ContainElement(PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":   Equal(field.ErrorTypeRequired),
+						"Field":  Equal("spec.strategy.rollingUpdate.maxSurge"),
+						"Detail": Equal("unable to convert maxSurge to int32"),
+					}))))
+				})
+			})
+
+			Context("Validate InPlaceUpdate strategy", func() {
+				BeforeEach(func() {
+					machineDeployment.Spec.Strategy = machine.MachineDeploymentStrategy{
+						Type: machine.InPlaceUpdateMachineDeploymentStrategyType,
+						InPlaceUpdate: &machine.InPlaceUpdateMachineDeployment{
+							UpdateConfiguration: machine.UpdateConfiguration{
+								MaxUnavailable: ptr.To(intstr.FromInt32(1)),
+								MaxSurge:       ptr.To(intstr.FromInt32(1)),
+							},
+							OrchestrationType: machine.OrchestrationTypeAuto,
+						},
+					}
+				})
+
+				It("should not return error if InPlaceUpdate strategy is valid", func() {
+					Expect(ValidateMachineDeployment(machineDeployment)).To(BeEmpty())
+				})
+
+				It("should return error if MaxUnavailable is not valid", func() {
+					machineDeployment.Spec.Strategy.InPlaceUpdate.UpdateConfiguration.MaxUnavailable = nil
+
+					Expect(ValidateMachineDeployment(machineDeployment)).To(ContainElement(PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":   Equal(field.ErrorTypeRequired),
+						"Field":  Equal("spec.strategy.inPlaceUpdate.maxUnavailable"),
+						"Detail": Equal("unable to convert maxUnavailable to int32"),
+					}))))
+				})
+
+				It("should return error if MaxSurge is not valid", func() {
```

Comment:
```suggestion
				It("should return error if MaxSurge is unspecified", func() {
```

Similar for other tests where values are set to `nil`